modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/web/internal/portlet/action/test/EditUserMVCActionCommandTest.java

```diff
@@ -76,81 +88,185 @@ public class EditUserMVCActionCommandTest {
+		String[] fieldEditableDomainFirstName = PropsUtil.getArray(
+			PropsKeys.FIELD_EDITABLE_DOMAINS, new Filter("firstName"));
+		String[] fieldEditableDomainPrefix = PropsUtil.getArray(
+			PropsKeys.FIELD_EDITABLE_DOMAINS, new Filter("prefix"));
+		String[] fieldEditableDomainSuffix = PropsUtil.getArray(
+			PropsKeys.FIELD_EDITABLE_DOMAINS, new Filter("suffix"));
+		String[] fieldEditableUserTypes = PropsValues.FIELD_EDITABLE_USER_TYPES;
 
 		try {
+			PropsUtil.set(
+				PropsKeys.FIELD_EDITABLE_USER_TYPES, StringPool.BLANK);
+			PropsUtil.set(
+				PropsKeys.FIELD_EDITABLE_DOMAINS + "[firstName]",
+				StringPool.STAR);
+			PropsUtil.set(
+				PropsKeys.FIELD_EDITABLE_DOMAINS + "[prefix]",
+				StringPool.BLANK);
+			PropsUtil.set(
+				PropsKeys.FIELD_EDITABLE_DOMAINS + "[suffix]",
+				StringPool.BLANK);
+
 			PrincipalThreadLocal.setName(TestPropsValues.getUserId());
 
 			PermissionChecker adminPermissionChecker =
 				PermissionCheckerFactoryUtil.create(TestPropsValues.getUser());
 
 			PermissionThreadLocal.setPermissionChecker(adminPermissionChecker);
 
-			_workflowDefinitionLinkLocalService.addWorkflowDefinitionLink(
-				null, TestPropsValues.getUserId(),
-				TestPropsValues.getCompanyId(),
-				GroupConstants.DEFAULT_LIVE_GROUP_ID, User.class.getName(), 0,
-				0, "Single Approver", 1);
+			Assert.assertTrue(
+				UsersAdminUtil.hasUpdateFieldPermission(
+					adminPermissionChecker, TestPropsValues.getUser(),
+					TestPropsValues.getUser(), "suffix"));
 
-			User user = _userLocalService.addUserWithWorkflow(
+			ListType prefixListType = _listTypeLocalService.getListType(
+				TestPropsValues.getCompanyId(), "dr",
+				ListTypeConstants.CONTACT_PREFIX);
+			ListType suffixListType = _listTypeLocalService.getListType(
+				TestPropsValues.getCompanyId(), "ii",
+				ListTypeConstants.CONTACT_SUFFIX);
+
+			User user = _userLocalService.addUser(
 				0, TestPropsValues.getCompanyId(), false, "test", "test", false,
 				RandomTestUtil.randomString(),
-				RandomTestUtil.randomString() + "@liferay.com", LocaleUtil.US,
+				RandomTestUtil.randomString() + "@example.com", LocaleUtil.US,
 				RandomTestUtil.randomString(), RandomTestUtil.randomString(),
-				RandomTestUtil.randomString(), 0, 0, true, 1, 1, 1970,
+				RandomTestUtil.randomString(), prefixListType.getListTypeId(),
+				suffixListType.getListTypeId(), true, 1, 1, 1970,
 				StringPool.BLANK, UserConstants.TYPE_REGULAR, null, null, null,
 				null, true,
 				ServiceContextTestUtil.getServiceContext(
 					TestPropsValues.getCompanyId(),
 					TestPropsValues.getGroupId(), TestPropsValues.getUserId()));
 
-			Assert.assertEquals(
-				WorkflowConstants.STATUS_PENDING, user.getStatus());
+			PrincipalThreadLocal.setName(user.getUserId());
+
+			PermissionChecker userPermissionChecker =
+				PermissionCheckerFactoryUtil.create(user);
+
+			PermissionThreadLocal.setPermissionChecker(userPermissionChecker);
```

Comment:
can you please check if the test can be simplified usign the `ContextUserReplace contextUserReplace` method?